### PR TITLE
Cli run lint fix

### PR DESCRIPTION
**Pull request by raymondfeng**
The PR adds `--format` option to run `npm lint:fix` to format generated code.

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/cli/lib/base-generator.js [hunk 1]
```diff
@@ -292,14 +313,17 @@ module.exports = class BaseGenerator extends Generator {
...
       debug(this.exitGeneration);
       this.log(chalk.red('Generation is aborted:', this.exitGeneration));
       // Fail the process
       process.exitCode = 1;
-      return false;
+      return;
+    }
+    if (this.options.format) {
+      this.log('Running npm run lint:fix to format the code...');
+      await this._runNpmScript(this.destinationRoot(), ['run', 'lint:fix']);
```
Comment (virkt25) [inline packages/cli/lib/base-generator.js]: Should there be a check to see if the script exists? Or does `_runNpmScript` do that under the hood?

Comment (raymondfeng) [reply packages/cli/lib/base-generator.js]: If the script does not exist, an error will be reported.

**Comment by raymondfeng** [comment]
> We should have some test cases here, some examples being npm run lint:fix being successfully run or package.json not having npm run lint:fix

I have created a test but it takes significant amount time to run:

1. It triggers `npm install` on the generated LB4 app, which might take up to 1 min
2. It runs `npm run lint:fix`

Adding 1 min to our build is painful. Do we have options to only run the test conditionally without slowing down CI?

**Comment by virkt25** [comment]
> Adding 1 min to our build is painful.

I don't think this is something that needs test coverage if it's been manually tested once. The extra time isn't worth the coverage for this option IMO. 

> Do we have options to only run the test conditionally without slowing down CI?

We can skip the test on CI by checking for `process.env.CI` (think that works on both Travis and AppVeyor) but like I said above I personally don't think the benefit is worth the coverage for this ... it's a CI option ... behind a flag ... that will either run or error out.

**Comment by shimks** [comment]
^^^ +1

**Comment by raymondfeng** [comment]
I removed `-f` to avoid confusion and added the logic to check if `lint:fix` script exists.

File: packages/cli/test/integration/generators/app.integration.js [hunk 2]
```diff
@@ -74,3 +74,26 @@ describe('app-generator with --applicationName', () => {
     );
   });
 });
+
+// The test takes about 1 min to install dependencies
+describe.skip('app-generator with --format', () => {
```
Comment (b-admike) [inline packages/cli/test/integration/generators/app.integration.js]: is the `skip` here intentional (I thought we wanted to disable on CI, but run locally)? Or do we want to enable it only when debugging it locally?

File: packages/cli/lib/base-generator.js [hunk 3]
```diff
@@ -310,6 +309,24 @@ module.exports = class BaseGenerator extends Generator {
...
+      if (pkg.scripts && pkg.scripts['lint:fix']) {
+        this.log("Running 'npm run lint:fix' to format the code...");
+        await this._runNpmScript(this.destinationRoot(), [
+          'run',
+          '-s',
+          'lint:fix',
+        ]);
+      } else {
+        this.log(
+          chalk.red("No 'lint:fix' script is configured in package.json."),
```
Comment (shimks) [inline packages/cli/lib/base-generator.js]: We should also test this section of the if ... else ... statement

**Pull request merged**